apps/oauth/tables.py

```diff
@@ -0,0 +1,50 @@
+import django_tables2 as tables
+from django.conf import settings
+from django.urls import reverse
+from django_tables2 import columns
+
+from apps.generics import actions
+from apps.generics.tables import TimeAgoColumn
+from apps.oauth.models import OAuth2Application
+
+
+def _update_url_factory(url_name, request, record, value):
+    """Factory for update URL."""
+    return reverse("oauth2_provider:application_edit", args=[record.pk])
+
+
+def _delete_url_factory(url_name, request, record, value):
+    """Factory for delete URL."""
+    return reverse("oauth2_provider:application_delete", args=[record.pk])
```

Comment:
you can use `url_name` to avoid having to specify it twice
```suggestion
    return reverse(url_name, args=[record.pk])


def _delete_url_factory(url_name, request, record, value):
    """Factory for delete URL."""
    return reverse(url_name, args=[record.pk])
```